### DISCUSSION TITLE
Mitsubishi AC retain status

### DISCUSSION THREAD

**Discussion by host-felice** [arendst/Tasmota#21783] [Q&A]
Hi everyone,

I can’t find the command to set the retain flag for the topic tele/********/HVACSETTINGS.

Every time I restart Home Assistant, the air conditioners status shows as "unknown" until Tasmota sends an update. I’ve tried all the retain commands I found on the Tasmota wiki without success.

Any help?

Thank you!

**Comment by sfromis** [comment]
No retain settings available there, but you could replicate the payload with an additional publish using retain, by having a small bot somewhere. If your Tasmota device is based on ESP32, this is easy to do with Berry on the same device. With "classic" rules, you'd have to have triggers picking up fields, one by one, before doing a new `Publish2`.

An alternate strategy, and probably the easiest, would be if you can send some IR command resulting in the settings being sent by the AC again. Not using HA, but I'd expect it to be possible to do something when it has restarted.

**Comment by host-felice** [comment]
no IR command, it works with serial connection. The problem is not to send a command, but to restore the previous state or retrive the actual state.

If I replicate the payload with the retain on, in whatever way I do it, I will end up with a double payload everytime. Unless I create another topic for every climate.

I can actually survive without the retain, I can somehow find a workaround to deal with, but it would be so easy to be able to set up a retain in that topic or a possibility to retrive the state like an empty topic on cmnd/tasmotas/state does.

**Reply by Jason2866** [reply]
Imho using retain to set a state is a bad idea. Could be correct or not. Before doing guess work via retain about a state, querying the device and using the delivered state is a way better approach.
Not using HA don't know how to do there. Using OpenHab, after (re)starting OpenHab a start up rule in OpenHab is querying all devices and "asking" there current real state.

In other words no change in Tasmota code for guessing a state.

**Reply by host-felice** [reply]
> querying the device and using the delivered state is a way better approach. Not using HA don't know how to do there. Using OpenHab, after (re)starting OpenHab a start up rule in OpenHab is querying all devices and "asking" there current real state.

how to you ask the state of the device? Since it works with mqtt, it doesn't matter if it is home assistant or openhab. I just need the topic and the command to send and I will create the automation to do it :-)

I know only the empty payload on cmnd/tasmotas/state but it doesn't retrive the state of the hvac.

**Reply by sfromis** [reply]
When you connect to another device (in this case HVAC), Tasmota does not maintain a state, and thus you cannot ask Tasmota. You'd have to ask the HVAC device about it's state, or do something which has that side effect.

**Reply by host-felice** [reply]
> When you connect to another device (in this case HVAC), Tasmota does not maintain a state, and thus you cannot ask Tasmota. You'd have to ask the HVAC device about it's state, or do something which has that side effect.

I guess a "teleperiod" will do the job. It will force to send every "tele" topic including HVACSETTINGS and the ACs will update their states on HA. So, an automation that fires on Home Assistant start will be ok.

Thanks for your help! If you think there is a better way, just let me know!

**Reply by sfromis** [reply]
> I guess a "teleperiod" will do the job. It will force to send every "tele" topic including HVACSETTINGS and the ACs will update their states on HA.

No, not at all. HVACSETTINGS is **not** telemetry data, even if it is published under the `tele` prefix. Telemetry data is only data which are maintained internally in Tasmota, like a local temperature sensor, or Wifi signal strength.

Again, Tasmota does not have **any** state of HVAC devices, and the only way of getting that is if you can make the HVAC device send state info, where Tasmota could relay that state info via MQTT.

**Reply by host-felice** [reply]
> No, not at all. HVACSETTINGS is **not** telemetry data, even if it is published under the `tele` prefix. Telemetry data is only data which are maintained internally in Tasmota, like a local temperature sensor, or Wifi signal strength.

Telemetry is the last known state of AC, isn't it?

The ACs send their data on tele/ACn/SENSOR and tele/ACn/HVACSETTINGS

tele/AC1/SENSOR = {"Time":"2024-07-16T17:05:08","MiElHVAC":{"Temperature":27.0,"Operation":"OFF","Compressor":"ON","roomtemp":"0300001100D0B6B6FE42000278B20000","status":"060000000002E4571B00004200000000","stage":"09000000000000000000000000000000"},"TempUnit":"C"}

It just say it is on, not the heat-dry-cool-fan_only state nor the fan speed, nor the swing mode

tele/AC1/HVACSETTINGS = {"Power":"ON","Mode":"cool","HAMode":"cool","Temp":26.0,"FanSpeed":"auto","SwingV":"auto","SwingH":"C","Bytes":"0200000103050000000083B446000000"}

I can check all the latest operation mode: it is on, in cool mode, temperature setpoint 26, fan speed auto, vertical swing auto.

If I change something with the remote, it immediately send an update on tele/AC1/HVACSETTINGS

Don't you think it is doing the job? I actually tried to reboot my Home Assistant, including the mqtt server. When I saw Tasmota was not connected to the mqtt server anymore, I turned off one AC and changed the setpoint temperature in the other ACs. When Home Assistant returned online, it sent an empty teleperiod and I got immediately the OFF status in that AC and all the other AC where correct setpoing temperature, the last one I changed while the mqtt server was off.
